### DISCUSSION TITLE
Trainer flags: amp_level vs. precision

### DISCUSSION THREAD

**Discussion by dianemarquette** [Lightning-AI/pytorch-lightning#8923] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi! 

I'm a bit confused regarding the trainer's flags. According to Lightning's [documentation](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#amp-level), the default settings are:
- `amp_level='O2'`  (i.e., "Almost FP16" Mixed Precision, cf. Nvidia [documentation](https://nvidia.github.io/apex/amp.html#opt-levels))
- `precision=32`

Isn't it a bit contradictory ? Is the default training mode full precision or mixed precision? 

Thanks in advance for your clarification :)

**Comment by tchaton** [comment, marked as answer]
Dear @dianemarquette,

`precision=32` is the default one. However, if you turn on `precision=16` and set `amp_backend="apex"`, then `amp_level=02` is considered as the default one.

Best,
T.C

**Reply by dianemarquette** [reply]
Thanks @tchaton for the quick and clear answer :)
